Makefile.doc

```diff
@@ -133,8 +148,9 @@ doc/common/version.tex: config/Makefile
 
 ### Changelog
 
-doc/unreleased.rst: $(wildcard doc/changelog/00-title.rst doc/changelog/*/*.rst)
+$(DOC_BUILD_DIR)/unreleased.rst: $(wildcard doc/changelog/00-title.rst doc/changelog/*/*.rst)
```

Comment:
What is the point of doing this out of tree if it is only used by the Sphinx build which is done in tree?

Reply:
Indeed, I can defer this change to a possible sphinx out-of-tree PR; I did setup that in the hope of advancing a bit more on that front, but only did the stdlib part.

Reply:
As you want. I can also merge as is.

Reply:
Let's merge as-is if you want, and also tweak the rest of targets so we bulid the full doc hygienically too.